plugin/src/main/java/io/jenkins/plugins/casc/SchemaGeneration.java

```diff
@@ -26,58 +33,73 @@ public static JSONObject generateSchema() {
          * The initial template for the JSON Schema
          */
         JSONObject schemaObject = new JSONObject(schemaTemplateObject.toString());
-        /**
-         * This generates the schema for the root configurators
-         * Iterates over the root elements and adds them to the schema.
-         */
-        schemaObject.put("properties", generateRootConfiguratorObject());
         /**
          * This generates the schema for the base configurators
          * Iterates over the base configurators and adds them to the schema.
          */
         JSONObject schemaConfiguratorObjects = new JSONObject();
-        ConfigurationAsCode configurationAsCode = ConfigurationAsCode.get();
-        for (Object configuratorObject : configurationAsCode.getConfigurators()) {
-            if (configuratorObject instanceof BaseConfigurator) {
-                BaseConfigurator baseConfigurator = (BaseConfigurator) configuratorObject;
-                List<Attribute> baseConfigAttributeList = baseConfigurator.getAttributes();
-                if (baseConfigAttributeList.size() == 0) {
-                    schemaConfiguratorObjects
-                        .put(((BaseConfigurator) configuratorObject).getTarget().getSimpleName().toLowerCase(),
-                            new JSONObject()
-                                .put("type", "object")
-                                .put("properties", new JSONObject()));
-
-                } else {
-                    JSONObject attributeSchema = new JSONObject();
-                    for (Attribute attribute : baseConfigAttributeList) {
-                        if (attribute.multiple) {
-                            generateMultipleAttributeSchema(attributeSchema, attribute);
-                        } else {
-                            if (attribute.type.isEnum()) {
-                                generateEnumAttributeSchema(attributeSchema, attribute);
+        DefaultConfiguratorRegistry registry = new DefaultConfiguratorRegistry();
+        final ConfigurationContext context = new ConfigurationContext(registry);
+
+        for (RootElementConfigurator rootElementConfigurator : RootElementConfigurator.all()) {
+            Set<Object> elements = new LinkedHashSet<>();
+            listElements(elements, rootElementConfigurator.describe(), context);
+
+            JSONObject rootConfiguratorProperties = new JSONObject();
+            rootConfiguratorProperties.put("type", "object")
+                .put("additionalProperties", false)
```

Comment:
@timja The `additionalProperties:false` is just being added for the root configurators, If wanted they could be added to every node.

Reply:
should be added to every node I think

Reply:
Noted :+1: